scripts/composer/ScriptHandler.php

```diff
@@ -88,6 +88,77 @@ public static function downloadDevelopPackages(Event $event) {
     }
   }
 
+  /**
+   * Reset local repositories to the default branch.
+   *
+   * @param \Composer\Script\Event $event
+   *   The script event.
+   */
+  public static function resetLocalRepositories(Event $event) {
+    $io = $event->getIO();
+    $repositoriesInfo = self::getLocalRepositoriesInfo($event);
+    $drupalFinder = new DrupalFinder();
+    $drupalFinder->locateRoot(getcwd());
+    $composerRoot = $drupalFinder->getComposerRoot();
+
+    $io->write(PHP_EOL);
+    foreach ($repositoriesInfo as $key => $info) {
+      $localBranch = trim(shell_exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' rev-parse --abbrev-ref HEAD'));
+
+      if ($localBranch === $info['branch']) {
+        $io->write($info['package'] . ' is already on branch ' . $info['branch'], TRUE);
+        continue;
+      }
+
+      $gitStatus = shell_exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' status --porcelain --untracked-files=no');
+      if (!empty($gitStatus)) {
+        $io->write('Stash local changes in ' . $info['package'] . ':' . $localBranch, TRUE);
+        exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' stash');
+      }
+
+      $io->write('Checkout ' . $info['package'] . ':' . $info['branch'], TRUE);
+      exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' checkout -q ' . $info['branch']);
+      exec('git -C ' . $composerRoot . '/' . $info['install_path'] . ' pull -q');
+    }
+  }
+
+  /**
+   * Collect information about local repositories.
+   *
+   * Retrieve available informazion about the repositories defined in the
+   * local-develop-packages key of the composer.json.
+   *
+   * @param \Composer\Script\Event $event
+   *   The script event.
+   *
+   * @return array
+   *   The collected repositories.
+   */
+  protected static function getLocalRepositoriesInfo(Event $event) {
+    $repositoriesInfo = [];
+    $composer = $event->getComposer();
+    $rootExtra = $composer->getPackage()->getExtra();
+    $packages = $rootExtra['local-develop-packages'];
+
+    foreach ($packages as $packageString => $packageVersion) {
+      $package = $composer->getRepositoryManager()->findPackage($packageString, $packageVersion);
+      if (!$package) {
+        continue;
+      }
+
+      $repository = $package->getRepository();
+      if ($gitDriver = $repository->getDriver()) {
+        $info = [];
+        $info['package'] = $packageString;
+        $info['install_path'] = self::getInstallPath($package, $composer);
+        $info['url'] = $gitDriver->getUrl();
+        $info['branch'] = (0 === strpos($packageVersion, 'dev-')) ? substr($packageVersion, strlen('dev-')) : '';
```

Comment:
I'm already thinking how to miss use this command. Can we have support for releases here not only `dev-`. ☺️

Reply:
Are you talking about checking out tags?

Reply:
Let#s not do this now. Should be another feature request